sys/stdio_uart/stdio_uart.c

```diff
@@ -87,6 +87,22 @@ ssize_t stdio_read(void* buffer, size_t count)
 
 ssize_t stdio_write(const void* buffer, size_t len)
 {
+#ifdef MODULE_STDIO_ETHOS
+    ethos_send_frame(&ethos, (const uint8_t *)buffer, len, ETHOS_FRAME_TYPE_TEXT);
+#else
+  #ifdef MODULE_STDIO_UART_ONLCR
+    const uint8_t *buf = (const uint8_t *)buffer;
+    const uint8_t cr = '\r';
+    size_t rem = len;
+    while (rem--) {
```

Comment:
If we ever get DMA on the UART it would be faster to write whole chunks - so better search for the `\n` with `strchr`/`memchr`

Reply:
That patch is fully functioning and tested ;)